### PR TITLE
Add fictitious flag to node metadata in network area diagram

### PR DESCRIPTION
**Pull request by massimo-ferraro**
**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
no

**What kind of change does this PR introduce?**
feature

**What is the current behavior?**
Metadata in network area diagram does not include the information about fictitious nodes

**What is the new behavior (if this is a feature change)?**
Diagram metadata includes the information about fictitious nodes

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

### CODE REVIEW

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/NodeMetadata.java [hunk 1]
```diff
@@ -35,4 +38,8 @@ public double getX() {
     public double getY() {
         return y;
     }
+
+    public boolean isFictitious() {
```
Comment (flo-dup) [inline network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/NodeMetadata.java:42]:
```suggestion
    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
    public boolean isFictitious() {
```

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/DiagramMetadata.java [hunk 2]
```diff
@@ -107,7 +108,8 @@ public DiagramMetadata addMetadata(Graph graph) {
                 getPrefixedId(node.getDiagramId()),
                 node.getEquipmentId(),
                 round(node.getX()),
-                round(node.getY()))));
+                round(node.getY()),
+                node instanceof VoltageLevelNode vlNode && vlNode.isFictitious())));
```
Comment (flo-dup) [inline network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/DiagramMetadata.java:112]: Could you move fictitious to Node interface and AbstractNode class?
```suggestion
                node.isFictitious())));
```

**Review by flo-dup — CHANGES_REQUESTED** [review]

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/NodeMetadata.java (hunk 1 above)
Comment (massimo-ferraro) [reply network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/NodeMetadata.java:43]: annotation added

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/DiagramMetadata.java (hunk 2 above)
Comment (massimo-ferraro) [reply network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/DiagramMetadata.java]: method isFictitious moved to Node interface

**Review by flo-dup — APPROVED** [review]

**Pull request merged**